dali/operators/math/expressions/expression_impl_gpu.cuh

```diff
@@ -66,16 +80,20 @@ __device__ void ExecuteBinOp(Result *result, Left l, const Right *r, int64_t ext
 template <ArithmeticOp op, typename Result, typename Left, typename Right>
 __device__ void ExecuteBinOp(Result *result, const Left *l, Right r, int64_t extent) {
   using meta = arithm_meta<op, GPUBackend>;
+  result += blockIdx.x * blockDim.x + threadIdx.x;
+  l += blockIdx.x * blockDim.x + threadIdx.x;
   for (int64_t i = blockIdx.x * blockDim.x + threadIdx.x; i < extent; i += blockDim.x * gridDim.x) {
-    result[i] = meta::impl(l[i], r);
+    *result = meta::impl(*l, r);
```

Comment:
Why? Does it help in any way?

Reply:
It seems so with smaller inputs, with bigger ones it doesn't make much difference.

Reply:
In that case, can we store the offset and step in variables to avoid repeating?

Reply:
Done I guess.